### PR TITLE
Wizard: Add Hostname functionality (HMS-5175)

### PR DESCRIPTION
**Pull request by regexowl**
This adds a validated hostname input and new tests.<br/><br/>JIRA: [HMS-5175](https://issues.redhat.com/browse/HMS-5175)

### CODE REVIEW

**Comment by regexowl** [comment]
Alright, let's give this a try -> /jira-epic HMS-5164

**Comment by regexowl** [comment]
What am I doing wrong with the epic? :thinking:

File: src/Components/CreateImageWizard/utilities/useValidation.tsx [hunk 1]
```diff
@@ -133,6 +137,20 @@ export function useFirstBootValidation(): StepValidation {
   };
 }
 
+export function useHostnameValidation(): StepValidation {
+  const hostname = useAppSelector(selectHostname);
+
+  if (!isHostnameValid(hostname)) {
+    return {
+      errors: {
+        hostname: 'Invalid hostname',
```
Comment (lucasgarfield) [inline src/Components/CreateImageWizard/utilities/useValidation.tsx:146]: Let's provide some additional guidance to users about what constitutes a valid hostname.

Had to look this up myself, of course. Run `man 5 hostname` and you get this:
```
HOSTNAME(5)                                                                         hostname                                                                         HOSTNAME(5)

NAME
       hostname - Local hostname configuration file

SYNOPSIS
       /etc/hostname

DESCRIPTION
       The /etc/hostname file configures the name of the local system. Unless overridden as described in the next section, systemd(1) will set this hostname during boot using
       the sethostname(2) system call.

       The file should contain a single newline-terminated hostname string. Comments (lines starting with a "#") are ignored. The hostname should be composed of up to 64 7-bit
       ASCII lower-case alphanumeric characters or hyphens forming a valid DNS domain name. It is recommended that this name contains only a single label, i.e. without any
       dots. Invalid characters will be filtered out in an attempt to make the name valid, but obviously it is recommended to use a valid name and not rely on this filtering.

       You may use hostnamectl(1) to change the value of this file during runtime from the command line. Use systemd-firstboot(1) to initialize it on mounted (but not booted)
       system images.
```

So I would suggest the error say: "Invalid hostname. The hostname should be composed of up to 64 7-bit
       ASCII lower-case alphanumeric characters or hyphens forming a valid DNS domain name. It is recommended that this name contains only a single label, i.e. without any
       dots. Invalid characters will be filtered out in an attempt to make the name valid, but obviously it is recommended to use a valid name and not rely on this filtering."

File: src/Components/CreateImageWizard/validators.ts [hunk 2]
```diff
@@ -90,3 +90,12 @@ export const isNtpServerValid = (ntpServer: string) => {
     /^([a-z0-9-]+)?(([.:/]{1,3}[a-z0-9-]+)){1,}$/.test(ntpServer)
   );
 };
+
+export const isHostnameValid = (hostname: string) => {
+  if (hostname !== '') {
+    return /^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9-]*[A-Za-z0-9])$/.test(
```
Comment (lucasgarfield) [inline src/Components/CreateImageWizard/validators.ts:96]: I think this expression fits what I'm proposing in my other comment. We just need to also set the max length to 64.

**Comment by lucasgarfield** [comment]
I added a commit with my suggested changes, hope you don't mind. Just want to keep the review cycle to a minimum so we can ship these before end of Q4.

**Review by lucasgarfield — APPROVED** [review]
:teapot:

**Comment by lucasgarfield** [comment]
Unit tests failing in pr_check but passing in dev-check. Admin merging.

18:51:03  FAIL  src/test/Components/CreateImageWizard/steps/ImageOutput/ImageOutput.test.tsx > Step Image output > revisit step button on Review works****

**Pull request merged**